UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/FunctionalTests/D365HttpPayloadValidationTest.cs

```diff
@@ -14,32 +14,53 @@ class D365HttpPayloadValidationTest
         private EnsApiClient EnsApiClient { get; set; }
         private TestConfiguration TestConfig { get; set; }
         private D365Payload D365Payload { get; set; }
+        private string EnsToken { get; set; }
 
         [SetUp]
-        public void Setup()
+        public async Task SetupAsync()
         {
             TestConfig = new TestConfiguration();
             EnsApiClient = new EnsApiClient(TestConfig.EnsApiBaseUrl);
 
             string filePath = Path.Combine(Directory.GetCurrentDirectory(), TestConfig.PayloadFolder, TestConfig.PayloadFileName);
 
             D365Payload = JsonConvert.DeserializeObject<D365Payload>(File.ReadAllText(filePath));
```

Comment:
As you've made the method `async` you can now use the `async` overload of `File.ReadAllText`:
```suggestion
            D365Payload = JsonConvert.DeserializeObject<D365Payload>(await File.ReadAllTextAsync(filePath));
```

Reply:
Done required changes.